.github/workflows/release-update-cloudfoundry-index.yml

```diff
@@ -0,0 +1,37 @@
+# This action updates the CloudFoundry java-buildpack release index after each release.
+# See https://github.com/cloudfoundry/java-buildpack/blob/main/docs/extending-repositories.md
+# Prerequisite: the repo must have a branch named "cloudfoundry".
+
+name: Update cloudfoundry release index
+on:
+  release:
+    types: [released]
+  workflow_dispatch:
+
+permissions:
+  contents: write
+
+jobs:
+
+  update-cloudfoundry-index-yml:
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.create-github-release.outputs.version }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: 'cloudfoundry'
```

Comment:
Hmm, I think this can be configured to avoid PRs; we already have a rule for `gh-pages` and the overheads test process is able to commit directly to that branch.

Reply:
If that works, then no objection.

Reply:
@mateuszrzeszutek Can you create the `cloudfoundry` branch?  I suggest you do this locally:
```
git checkout --orphan cloudfoundry
touch README.md
... commit
git push cloudfoundry
```

This will create a new branch with no history, which is what we want for this branch.
If you want to add something to the README, go ahead, otherwise I can submit a PR once the branch is in place.

Reply:
I created the branch.

`gh-pages` completely bypasses any protections: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/.github/repository-settings.md#gh-pages

I'm not sure if that's what we want for this or not, since it will essentially contain production code(?)

may be worth discussing in Thu SIG

Reply:
Alternatively we could just create PRs for that; 1 PR a month to accept is not much.

Reply:
Ok, I've updated it to create a PR instead of pushing directly to the repo branch. @trask I suggest you add branch protections if you haven't done so already.